core/src/test/java/org/apache/druid/data/input/HandlingInputRowIteratorTest.java

```diff
@@ -81,58 +87,81 @@ public void throwsExceptionWhenYieldingNext()
 
   public static class PresentRowTest
   {
+    // Create variables for tracking behaviors of mock object
+
+    boolean unsuccessfulHandlerSuccessful;
+
+    // Create variables for tracking behaviors of mock object
+
+    boolean successfulHandlerSuccessful;
+
     private static final InputRow INPUT_ROW1 = EasyMock.mock(InputRow.class);
     private static final InputRow INPUT_ROW2 = EasyMock.mock(InputRow.class);
     private static final List<InputRow> INPUT_ROWS = Arrays.asList(INPUT_ROW1, INPUT_ROW2);
 
-    private TestInputRowHandler successfulHandler;
-    private TestInputRowHandler unsuccessfulHandler;
+    private InputRowHandler successfulHandler;
+    private InputRowHandler unsuccessfulHandler;
 
     @Before
     public void setup()
     {
-      successfulHandler = new TestInputRowHandler(true);
-      unsuccessfulHandler = new TestInputRowHandler(false);
+      // Construct mock object
+      successfulHandler = mock(HandlingInputRowIterator.InputRowHandler.class);
+      // Constructing variables that used for tracking behavior of the mocking
+      // object.
+      successfulHandlerSuccessful = true;
+      // Method Stubs
+      when(successfulHandler.handle(any(InputRow.class))).thenAnswer((stubInvo) -> {
+        return successfulHandlerSuccessful;
+      });
+      // Construct mock object
+      unsuccessfulHandler = mock(HandlingInputRowIterator.InputRowHandler.class);
+      // Constructing variables that used for tracking behavior of the mocking
+      // object.
+      unsuccessfulHandlerSuccessful = false;
+      // Method Stubs
+      when(unsuccessfulHandler.handle(any(InputRow.class))).thenAnswer((stubInvo) -> {
+        return unsuccessfulHandlerSuccessful;
+      });
     }
 
     @Test
     public void hasNext()
     {
       HandlingInputRowIterator target = createInputRowIterator(unsuccessfulHandler, unsuccessfulHandler);
       Assert.assertTrue(target.hasNext());
-      Assert.assertFalse(unsuccessfulHandler.invoked);
+      Mockito.verify(unsuccessfulHandler, Mockito.never()).handle(any(InputRow.class));
     }
 
     @Test
     public void yieldsNextIfUnhandled()
     {
       HandlingInputRowIterator target = createInputRowIterator(unsuccessfulHandler, unsuccessfulHandler);
       Assert.assertEquals(INPUT_ROW1, target.next());
-      Assert.assertTrue(unsuccessfulHandler.invoked);
+      Mockito.verify(unsuccessfulHandler, Mockito.atLeastOnce()).handle(any(InputRow.class));
```

Comment:
I think we want to verify that `INPUT_ROW1` was passed to the handle object
```suggestion
      Mockito.verify(unsuccessfulHandler, Mockito.atLeastOnce()).handle(INPUT_ROW1);
```

Reply:
Why `atLeastOnce()` - is this really called multiple times? Similar question for other uses of `atLeastOnce` in this test

Reply:
I agree with you, by changing `atLeastOnce()` to `times(int)` can make test more powerful. I use `atLeastOnce()` instead of `times(1)` here is because the original test is using boolean to verify the method behavior. Boolean can only be used to check whether the method is been executed or not instead of tracking the exact execution times. Also, in [yieldsNextIfUnhandled()](https://github.com/apache/druid/pull/11529/files/8dd713e6e547381ae24928f672d1561b6a670335#diff-c861ad661bc346cc59e01172af11c513b2092026fb1892fcf04d102a1394ef47R139), `handle()` executed twice since we passed two `unsuccessfulHandler` to `HandlingInputRowIterator`.